### ISSUE TITLE
Converting StringLiteralExpression to InterpolatedString requires me to handle special characters

### ISSUE THREAD

**Issue by paul1956** [dotnet/roslyn#12811]
**Version Used**: 
1.3.2
**Steps to Reproduce**:
```
Case SyntaxKind.StringLiteralExpression
                Dim InvicationRightToken As SyntaxToken = CType(invocation, LiteralExpressionSyntax).Token
                Dim InvicationRightTokenValueText As String = InvicationRightToken.ValueText.Replace("""", """""")
                Dim TextToken As SyntaxToken = SyntaxFactory.InterpolatedStringTextToken(InvicationRightToken.LeadingTrivia, InvicationRightTokenValueText, InvicationRightTokenValueText, InvicationRightToken.TrailingTrivia)
                Dim InterpolatedStringTextToken As InterpolatedStringTextSyntax = SyntaxFactory.InterpolatedStringText(TextToken)
                Return InterpolatedStringTextToken
 ```

**Expected Behavior**:
When converting from a StringLiteralExpression that contains embedded "'s to a InterpolatedStringTextSyntax I have to deal with spacial characters like "
**Actual Behavior**:
If I don't include .Replace("""", """""") and I am not sure what other characters I need to deal with the resulting  InterpolatedString get a compile error.

**Comment by gafter** [comment]
You do indeed require different characters inside an interpolated string token than would be required inside a normal string token. You have to escape `{`, and `}` by doubling them.

The quote character you need to escape the same way in an interpolated string as in a normal string.

@AnthonyDGreen Do you care to look at this?

**Comment by AnthonyDGreen** [comment]
Hey @paul1956,

I'm not sure what you're asking. Could you rephrase?

**Comment by paul1956** [comment]
@AnthonyDGreen if you look at the second line of the sample I need to basically double the number of quotes to make it work correctly (I don't deal with other special characters yet). I start with a StringLiteralExpression which might contain quotes (already doubled to get them in there) or brackets or anything else and I want to convert it to an InterpolatedStringTextToken. Why would I think I need to escape anything and how do I know what I need to escape if I have to do it manually. Even @gafter comment is confusing. The string as already quoted in a StringLiteralExpression so I presumed it was already escaped in an appropriate way.

**Comment by gafter** [comment]
The `.ValueText` of a string has had the escapes processed and removed. If you want the text of the original token, including the original escapes, you could just use `.ToString()` or (I think) `.Text`.
